lettermgmt/models/res_letter.py

```diff
@@ -1,165 +1,200 @@
+        help='Sent / Receive Source')
+
+    category_ids = fields.Many2many(
+        'letter.category',
+        string="Tags",
+        help="Classification of Document.")
+
+    folder_id = fields.Many2one(
+        'letter.folder',
+        string='Folder',
+        help='Folder which contains letter.')
+
+    type_id = fields.Many2one(
+        'letter.type',
+        string="Type",
+        help="Type of Letter, Depending upon size.")
+
+    weight = fields.Float(help='Weight (in KG)')
+    size = fields.Char(help='Size of the package.')
+
+    track_ref = fields.Char(
+        string='Tracking Reference',
+        help="Reference Number used for Tracking.")
+    orig_ref = fields.Char(
+        string='Original Reference',
+        help="Reference Number at Origin.")
+    expeditor_ref = fields.Char(
+        string='Expeditor Reference',
+        help="Reference Number used by Expeditor.")
+
+    parent_id = fields.Many2one(
+        'res.letter',
+        string='Parent',
+        groups='lettermgmt.group_letter_thread')
+    child_line = fields.One2many(
+        'res.letter',
+        'parent_id',
+        string='Letter Lines',
+        groups='lettermgmt.group_letter_thread')
+
+    reassignment_ids = fields.One2many(
+        'letter.reassignment',
+        'letter_id',
+        string='Reassignment lines',
+        help='Reassignment users and comments',
+        groups='lettermgmt.group_letter_reasignment')
+
+    # This field seems to be unused. TODO: Remove it?
+    extern_partner_ids = fields.Many2many(
+        'res.partner',
+        string='Recipients')
+
+    @api.model
+    def create(self, vals):
+        if ('number' not in vals) or (vals.get('number') in ('/', False)):
+            sequence = self.env['ir.sequence']
+            move_type = vals.get('move', self.env.context.get(
+                'default_move', self.env.context.get('move', 'in')))
+            vals['number'] = sequence.get('%s.letter' % move_type)
+        return super(ResLetter, self).create(vals)
 
-    def action_create(self, cr, uid, ids, context=None):
-        """Put the state of the letter into Crated"""
-        for letter in self.browse(cr, uid, ids, context=context):
-            self.write(
-                cr, uid, [letter.id], {'state': 'created'}, context=context)
+    @api.multi
+    def action_cancel(self):
+        """ Put the state of the letter into Cancelled """
+        self.write({'state': 'cancel'})
         return True
 
-    def action_validate(self, cr, uid, ids, context=None):
-        """Put the state of the letter into Validated"""
-        for letter in self.browse(cr, uid, ids, context=context):
-            self.write(
-                cr, uid, [letter.id], {'state': 'validated'}, context=context)
+    @api.multi
+    def action_cancel_draft(self):
+        """ Go from cancelled state to draf state """
+        self.write({'state': 'draft'})
+        self.delete_workflow()
+        self.create_workflow()
```

Comment:
At the time of this PR, I was just starting to learn odoo. I guess I've just copied this from another module's code, and it doesn't apply here at all.